### PR TITLE
OCD-4545 = Edit Cognito user information

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: src/app/api/users.jsx [hunk 1]
```diff
@@ -22,8 +22,19 @@ const usePutUser = () => {
...
+    },
+  });
+};
+
 export {
+  usePutUser, // eslint-disable-line import/prefer-default-export
+  usePutCognitoUser, // eslint-disable-line import/prefer-default-export
   usePostCreateCognitoInvitedUser,
   usePostCreateInvitedUser,
-  usePutUser,
```
Comment (andlar) [inline src/app/api/users.jsx]: These line don't need the eslint-disable. That's only if there's only a single export. Also, the standard in the API files is to sort the export alphabetically (as well as sorting the functions, but there's no change needed there)

File: src/app/components/user/cognito-user-edit.jsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+} from 'prop-types';
+import {
+  Card,
+  CardHeader,
+  CardContent,
+  FormControlLabel,
+  Switch,
+  ThemeProvider,
+  makeStyles,
+  Typography,
```
Comment (andlar) [inline src/app/components/user/cognito-user-edit.jsx]: These should be alphabetized

File: src/app/components/user/cognito-user-edit.jsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+} from '@material-ui/core';
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+
+import theme from '../../themes/theme';
+import { ChplTextField } from '../util';
+import { ChplActionBar } from '../action-bar';
+import {
+  user as userPropType,
+} from '../../shared/prop-types';
```
Comment (andlar) [inline src/app/components/user/cognito-user-edit.jsx]:
```suggestion
import theme from 'themes/theme';
import { ChplTextField } from 'components/util';
import { ChplActionBar } from 'components/action-bar';
import {
  user as userPropType,
} from 'shared/prop-types';
```

We don't do relative imports, except for "local relative". I usually sort imports by the values after the `from` keyword as well

File: src/app/components/user/cognito-user-edit.jsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+const useStyles = makeStyles(() => ({
```
Comment (andlar) [inline src/app/components/user/cognito-user-edit.jsx]:
```suggestion
const useStyles = makeStyles({
```

Making it a function was necessary when we had to pass some sort of value into the makeStyles call. We don't, so having it a function is overkill

File: src/app/components/user/cognito-user-edit.jsx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    onSubmit: () => {
+      save();
+    },
+    validationSchema,
+    validateOnChange: false,
+    validateOnMount: true,
+  });
+
+  return (
+    <ThemeProvider theme={theme}>
```
Comment (andlar) [inline src/app/components/user/cognito-user-edit.jsx]: If this component is used as a sub-component of any of our React ones, then somewhere up the tree there's already a ThemeProvider that provides the theme. Putting another one here is extraneous

File: src/app/components/user/cognito-user-view.jsx [hunk 6]
```diff
@@ -1,15 +1,23 @@
...
   Card,
+  CardActions,
   CardContent,
   CardHeader,
   ThemeProvider,
   Typography,
   makeStyles,
 } from '@material-ui/core';
 
+import { ChplTooltip } from '../util';
```
Comment (andlar) [inline src/app/components/user/cognito-user-view.jsx]:
```suggestion
import { ChplTooltip } from 'components/util';
```

File: src/app/components/user/users.jsx [hunk 7]
```diff
@@ -148,10 +168,10 @@ function ChplUsers({
         <ChplCognitoUserView
           key={userToDisplay.cognitoId}
           user={userToDisplay}
+          dispatch={handleDispatch}
         />
       );
-    }
-    if (userToDisplay.userId) {
+    } if (userToDisplay.userId) {
```
Comment (andlar) [inline src/app/components/user/users.jsx]:
```suggestion
    } 
    if (userToDisplay.userId) {
```

File: src/app/components/user/users.jsx [hunk 8]
```diff
@@ -163,6 +183,27 @@ function ChplUsers({
...
+  const displayUserEdit = (userToEdit) => {
+    if (userToEdit.cognitoId) {
+      return (
+        <ChplCognitoUserEdit
+          user={userToEdit}
+          errors={errors}
+          dispatch={handleDispatch}
+        />
+      );
+    } if (userToEdit.userId) {
```
Comment (andlar) [inline src/app/components/user/users.jsx]:
```suggestion
    } 
    if (userToEdit.userId) {
```

Either a newline, or an `else if`. Putting the if on the same line as the end of the previous block is odd

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

File: src/app/components/user/cognito-user-view.jsx [hunk 9]
```diff
@@ -87,6 +99,21 @@ function ChplCognitoUserView({ user: initialUser }) {
...
+                aria-label={`Edit ${user.fullName}`}
+                onClick={edit}
+              >
+                <EditOutlinedIcon />
+              </Button>
+            </ChplTooltip>
+          </ButtonGroup>
+        </CardActions>
       </Card>
     </ThemeProvider>
```
Comment (andlar) [inline src/app/components/user/cognito-user-view.jsx]: This doesn't need to have the `ThemeProvider` tag, as some parent of the component should have it. That should mean that the import of the `theme` can be removed too. I know this isn't specifically related to the changes for this ticket, but could be considered part of the gradual clean-up we're supposed to keep doing

File: src/app/components/user/cognito-user-view.jsx [hunk 10]
```diff
@@ -1,15 +1,22 @@
...
+  CardActions,
   CardContent,
   CardHeader,
-  ThemeProvider,
   Typography,
   makeStyles,
 } from '@material-ui/core';
 
+import { ChplTooltip } from 'components/util';
 import theme from 'themes/theme';
```
Comment (andlar) [inline src/app/components/user/cognito-user-view.jsx]: Without the `<ThemeProvider>`, unless the `theme` is used in the styles thing (usually for page size issues) this import probably isn't needed

**Review by kekey1 — APPROVED** [review]

**Review by andlar — APPROVED** [review]

**Pull request merged**